include/z64.h

```diff
@@ -969,6 +969,77 @@ typedef struct {
     /* 0x4C */ u32 unk_4C;
 } PreRender; // size = 0x50
 
+#define TRANS_TRIGGER_OUT -20 // transition is going out
+#define TRANS_TRIGGER_OFF 0 // transition is not active
+#define TRANS_TRIGGER_IN 20 // activate transition to come in
+
+typedef enum {
+    /* 00 */ TRANS_MODE_OFF,
+    /* 01 */ TRANS_MODE_SETUP,
+    /* 02 */ TRANS_MODE_INSTANCE_INIT,
+    /* 03 */ TRANS_MODE_INSTANCE_RUNNING,
+    /* 04 */ TRANS_MODE_FILL_WHITE_INIT,
+    /* 05 */ TRANS_MODE_FILL_IN,
+    /* 06 */ TRANS_MODE_FILL_OUT,
+    /* 07 */ TRANS_MODE_FILL_BROWN_INIT,
+    /* 08 */ TRANS_MODE_08, // unused
+    /* 09 */ TRANS_MODE_09, // unused
+    /* 10 */ TRANS_MODE_INSTANT,
+    /* 11 */ TRANS_MODE_INSTANCE_WAIT,
+    /* 12 */ TRANS_MODE_SANDSTORM_INIT,
+    /* 13 */ TRANS_MODE_SANDSTORM,
+    /* 14 */ TRANS_MODE_SANDSTORM_END_INIT,
+    /* 15 */ TRANS_MODE_SANDSTORM_END,
+    /* 16 */ TRANS_MODE_CS_BLACK_FILL_INIT,
+    /* 17 */ TRANS_MODE_CS_BLACK_FILL
+} TransitionMode;
+
+typedef enum {
+    /* 00 */ TRANS_TYPE_WIPE,
+    /* 01 */ TRANS_TYPE_TRIFORCE,
+    /* 02 */ TRANS_TYPE_FADE_BLACK,
+    /* 03 */ TRANS_TYPE_FADE_WHITE,
+    /* 04 */ TRANS_TYPE_FADE_BLACK_FAST,
+    /* 05 */ TRANS_TYPE_FADE_WHITE_FAST,
+    /* 06 */ TRANS_TYPE_FADE_BLACK_SLOW,
+    /* 07 */ TRANS_TYPE_FADE_WHITE_SLOW,
+    /* 08 */ TRANS_TYPE_WIPE_FAST,
+    /* 09 */ TRANS_TYPE_FILL_WHITE2, 
+    /* 10 */ TRANS_TYPE_FILL_WHITE,
+    /* 11 */ TRANS_TYPE_INSTANT,
+    /* 12 */ TRANS_TYPE_FILL_BROWN,
+    /* 13 */ TRANS_TYPE_FADE_WHITE_CS_DELAYED,
+    /* 14 */ TRANS_TYPE_SANDSTORM_PERSIST,
+    /* 15 */ TRANS_TYPE_SANDSTORM_END,
+    /* 16 */ TRANS_TYPE_CS_BLACK_FILL,
+    /* 17 */ TRANS_TYPE_FADE_WHITE_INSTANT,
+    /* 18 */ TRANS_TYPE_FADE_GREEN,
+    /* 19 */ TRANS_TYPE_FADE_BLUE,
+    // transition types 20 - 31 are unused
+    // transition types 32 - 55 are constructed using the TRANS_TYPE_CIRCLE macro
+    /* 56 */ TRANS_TYPE_MAX = 56
```

Comment:
`TRANS_TYPE_MAX` is weird to begin with. 
I need to look at it closer but there are checks for being >= to 56, but in both cases it doesnt necessarily treat it as "invalid"

for the instance types, if the type is >= to 56 it will just end the transition and perform the scene load instantly. This is probably related to 
```
void func_800BC88C(GlobalContext* globalCtx) {
    globalCtx->transitionCtx.transitionType = -1;
}
```
pretty sure this is actually 0xFF not -1 since u8, and 255 would satisfy that condition of >= 56

(note that this does not have the same meaning as 0xFF for nextTransition. that should get a define of its own though, mainly wanted to figure out this usage first

Reply:
oh nvm, got the transitionCtx and global context versions of the variable confused. this one is s32 lol
in that case i dont know how these tie in to each other